### DISCUSSION TITLE
New Graphical Marks

### DISCUSSION THREAD

**Discussion by mattjvincent** [genome-spy/genome-spy#49] [General]
I am curious how hard/easy it would be to add line plots, whisker plots?

[https://vega.github.io/vega-lite/examples/line.html](url)

[https://vega.github.io/vega-lite/examples/boxplot_minmax_2D_vertical.html](url)

**Comment by tuner** [comment]
Hi Matt,

It's more about prioritization than difficulty. I've been lately busy with other features, mainly related to working with large patient cohorts, bookmarking, and sharing findings.

Anyway, at some point I'll continue improving the visualization capabilities. For instance, I need `"area"` mark for density visualizations. Maybe I could implement the `"line"` mark too, because they are technically quite similar to each other.

When it comes to box plots, it could actually be possible to specify them with the current feature set. Vega-Lite's box plot marks are, in fact, "composite" marks that layer a `"rule"`, "`rect`", and a `"tick"` mark. And there's also a transform that computes the statistics for the box plot. I recently implemented [stroked rectangles](https://genomespy.app/docs/grammar/mark/rect/) and the `"tick"` mark is actually a `"rule"` with a specific parameterization.

So, if the question was actually when to expect such features to be implemented, please add a new issue (or preferably two separate issues) with some motivational use cases :-)

**Comment by mattjvincent** [comment]
First, great work with adding a bunch of new features!  

Second, have you given any thoughts to adding in lines? Or other types? I'd like to use GenomeSpy for the charts in the attached image
<img width="600" alt="lines" src="https://github.com/genome-spy/genome-spy/assets/1842565/19b7bcd6-6c6f-43ec-9d97-7f16f2df22d1">
.

**Comment by tuner** [comment]
Thanks!

Unfortunately, your issue (#50) is still open, as we haven't had any pressing use cases for lines in our lab. I'd like to implement them soon (or get implemented) but I cannot promise anything for now.
